### PR TITLE
[Utils refactor] move Merge into polymorphic package

### PR DESCRIPTION
**Pull request by benoitgrelard**

### CODE REVIEW

File: packages/core/utils/src/typeUtils.ts [hunk 1]
```diff
@@ -1,17 +0,0 @@
-/**
```
Comment (benoitgrelard) [inline packages/core/utils/src/typeUtils.ts:1]: These weren't used anywhere.

File: packages/core/utils/src/types.ts [hunk 2]
```diff
@@ -1,8 +0,0 @@
-/**
- * This type is included in the DOM lib but is deprecated. It's still quite useful, so we want to
- * include it here and reference it when possible to avoid any issues when updating TS.
- */
-export type ElementTagNameMap = HTMLElementTagNameMap &
```
Comment (benoitgrelard) [inline packages/core/utils/src/types.ts:5]: Not used anywhere.

File: packages/core/utils/src/types.ts [hunk 3]
```diff
@@ -1,8 +0,0 @@
...
-  Pick<SVGElementTagNameMap, Exclude<keyof SVGElementTagNameMap, keyof HTMLElementTagNameMap>>;
-
-export type Merge<P1 = {}, P2 = {}> = Omit<P1, keyof P2> & P2;
```
Comment (benoitgrelard) [inline packages/core/utils/src/types.ts:8]: Moved into Polymorphic.

File: packages/react/toolbar/src/Toolbar.tsx [hunk 4]
```diff
@@ -4,22 +4,21 @@ import { Primitive } from '@radix-ui/react-primitive';
...
 import * as RadioGroupPrimitive from '@radix-ui/react-radio-group';
 
+import type * as Polymorphic from '@radix-ui/react-polymorphic';
+
 /* -------------------------------------------------------------------------------------------------
  * Toolbar
  * -----------------------------------------------------------------------------------------------*/
 
 const TOOLBAR_NAME = 'Toolbar';
-const TOOLBAR_CONTEXT_NAME = TOOLBAR_NAME + 'Context';
```
Comment (benoitgrelard) [inline packages/react/toolbar/src/Toolbar.tsx:17]: Leftover from context PRs I think

File: packages/react/toolbar/src/Toolbar.tsx [hunk 5]
```diff
@@ -4,22 +4,21 @@ import { Primitive } from '@radix-ui/react-primitive';
 import { Slot } from '@radix-ui/react-slot';
 import { getSelector } from '@radix-ui/utils';
 import { createContextObj, composeEventHandlers } from '@radix-ui/react-utils';
-import type * as Polymorphic from '@radix-ui/react-polymorphic';
-import type { Merge } from '@radix-ui/utils';
 import { Separator as SeparatorPrimitive } from '@radix-ui/react-separator';
 import * as RadioGroupPrimitive from '@radix-ui/react-radio-group';
 
+import type * as Polymorphic from '@radix-ui/react-polymorphic';
```
Comment (jjenzz) [inline packages/react/toolbar/src/Toolbar.tsx:10]: 👍

**Review by jjenzz — APPROVED** [review]

**Pull request merged**